### DISCUSSION TITLE
Load Balancing - Circuits

### DISCUSSION THREAD

**Discussion by JanVanr** [marq24/ha-evcc#32] [Q&A]
Any idea how I can set the maxPower for a circuit from HA?

As we get billed for the Max peak consumption in a month, I'd like to limit what evcc can 'consume' based on the max peak of the month.

Using the below in my evcc.yaml works great, but needs manual adjustments throughout the month to limit the max power to the previous peak.

From the evcc.io site: (load management)
circuits:
  - name: main # if there is only one circuit defined the name needs to be 'main'
    title: 'main circuit' # name for the UI (not implemented in UI yet)
    maxCurrent: 63 # 63A (optional)
    maxPower: 30000 # 30kW (optional)
    meter: grid # optional

loadpoints:
  - title: Garage
    circuit: main

**Comment by marq24** [comment]
you might be able in your evcc circuits configuration to make use of an 'external' source - like I have shown hot it can be done for meters (https://github.com/marq24/ha-evcc/blob/main/HA_AS_EVCC_SOURCE.md) - WHEN this is possible for `circuits` as well, then you might be able to feed the `maxPower` attribute from a entity in HA (this can be e.g. a simple text_input/number_input helper)...

That's the only way how you might can control this - but even if you can feed the data to this evcc.config you still might need a evcc restart with every change... 

IMHO this is more a question for the evcc developers - then the HA integration...

**Comment by JanVanr** [comment]
Tried the http GET as you described, it works great for meters but I cannot get it to work for circuits.
As someone asked a similar question on the evcc GitHub, I will keep an eye on that discussion.

**Reply by JanVanr** [reply]
This has been fixed in evcc.
```

circuits:
  - name: main
    maxPower: 6500
    GetMaxPower: 
      source: mqtt
      topic: sim/circuit/maxpower
    meter: grid
    parent:

```
